windows/windows_update_install/install_msu_file.yml

```diff
@@ -1,10 +1,31 @@
 # Copyright 2023-2024 VMware, Inc.
 # SPDX-License-Identifier: BSD-2-Clause
 ---
+# Parameters:
+#   msu_file_path: the .msu file path in guest OS.
+#
+- name: "Get the .msu file basename"
+  include_tasks: ../utils/win_execute_cmd.yml
+  vars:
+    win_powershell_cmd: "(Get-Item {{ msu_file_path }}).BaseName"
+
+- name: "Set the .msu file basename"
+  ansible.builtin.set_fact:
+    msu_file_basename: "{{ win_powershell_cmd_output.stdout_lines[0] }}"
+  when:
+    - win_powershell_cmd_output.stdout_lines is defined
+    - win_powershell_cmd_output.stdout_lines | length == 1
+
+- name: "Check if the .msu file basename is set or not"
+  ansible.builtin.assert:
+    that:
+      - msu_file_basename is defined
+      - msu_file_basename
+    fail_msg: "Failed to get the .msu file basename."
```

Comment:
I thinks fail message can contain more info, e.g., "Failed to get the .msu file basename from specified path: {{ msu_file_path }}."

Reply:
Updated.